src/Soneta.Sdk/Sdk/Sdk.targets

```diff
@@ -18,6 +25,110 @@
     <RunArguments>$(StartArguments)</RunArguments>
   </PropertyGroup>
 
+  <UsingTask TaskName="CheckForSdkUpdates" TaskFactory="RoslynCodeTaskFactory" AssemblyFile="$(MSBuildBinPath)\Microsoft.Build.Tasks.Core.dll">
+    <ParameterGroup>
+      <GlobalJsonFilePath ParameterType="System.String" Required="true" />
+      <LatestVersionFilePath ParameterType="System.String" Required="true" />
+      <NugetApiUrl ParameterType="System.String" Required="true" />
+      <SdkName ParameterType="System.String" Required="true" />
+      <UpdateInterval ParameterType="System.Int32" Required="true" />
+      <IsUpdateAvailable ParameterType="System.Boolean" Output="true" />
+      <CurrentVersion ParameterType="System.String" Output="true" />
+      <LatestVersion ParameterType="System.String" Output="true" />
+    </ParameterGroup>
+
+    <Task>
+      <Using Namespace="System.Linq" />
+      <Using Namespace="System.Net.Http" />
+      <Using Namespace="System.Text.RegularExpressions" />
+      <Code Type="Method" Language="cs">
+        <![CDATA[
+        private const string GlobalJsonSdkVersionPattern = @"(?<=""Soneta.Sdk""\s*:\s*"")([0-9]+\.?){3,}(?=.*?"")";
+        private const string NugetApiLatestVersionPattern = @"(?<="")([0-9]+\.?){3,}(?=.*?""\s*])";
+        
+        public override bool Execute()
+        {
+            CurrentVersion = GetCurrentVersion();
+            LatestVersion = GetLatestVersion();
+            IsUpdateAvailable = CheckIfUpdateAvailable(CurrentVersion, LatestVersion);
+
+            return true;
+        }
+
+        private string GetCurrentVersion()
+        {
+            if (File.Exists(GlobalJsonFilePath))
+            {
+                var json = File.ReadAllText(GlobalJsonFilePath);
+                return GetMatchingValue(json, GlobalJsonSdkVersionPattern);
+            }
+
+            return null;
+        }
+
+        private string GetLatestVersion()
+        {
+            var lastWriteTime = File.GetLastWriteTime(LatestVersionFilePath);
+            var updateTime = lastWriteTime.AddMinutes(UpdateInterval);
+
+            if (updateTime < DateTime.Now)
+            {
+                var latestVersion = GetLastVersionFromApi();
+
+                using (var sw = File.CreateText(LatestVersionFilePath))
+                    sw.WriteLine(latestVersion);
+
+                return latestVersion;
+            }
+            else
+            {
+                return File.ReadAllText(LatestVersionFilePath).Trim();
+            }
+        }
+
+        private string GetLastVersionFromApi()
+        {
+            using (var httpClient = new HttpClient())
+            {
+                var json = httpClient.GetStringAsync(NugetApiUrl).Result;
+                return GetMatchingValue(json, NugetApiLatestVersionPattern);
+            }
+        }
+
+        private string GetMatchingValue(string json, string pattern)
```

Comment:
Wujek Bob podpowiada żeby przerzucić to pod GetCurrentVersion() :)

Reply:
Wujek Bob zwykle ma rację, załatwione.